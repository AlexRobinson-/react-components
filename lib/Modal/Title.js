'use strict';

exports.__esModule = true;

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _ThemedComponent2 = require('../utils/theming/ThemedComponent');

var _ThemedComponent3 = _interopRequireDefault(_ThemedComponent2);

var _styles = require('./styles');

var _styles2 = _interopRequireDefault(_styles);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Header = function (_ThemedComponent) {
  (0, _inherits3.default)(Header, _ThemedComponent);

  function Header(props, context) {
    (0, _classCallCheck3.default)(this, Header);
    return (0, _possibleConstructorReturn3.default)(this, _ThemedComponent.call(this, props, context, {
      namespace: 'Modal',
      styles: _styles2.default
    }));
  }

  Header.prototype.render = function render() {
    var children = this.props.children;
    var theme = this.theme;


    return _react2.default.createElement(
      'h1',
      { className: theme.header },
      children
    );
  };

  return Header;
}(_ThemedComponent3.default);

Header.propTypes = {
  children: _react.PropTypes.node.isRequired
};
exports.default = Header;