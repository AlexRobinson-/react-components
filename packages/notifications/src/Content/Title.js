import styled from 'styled-components';
import classNames from 'classnames';
import CalloutStyles from '@zendesk/garden-css-callouts/dist/index.css';
import { retrieveTheme } from 'garden-react-theming';

const Title = styled.div.attrs({
  className: classNames(CalloutStyles['c-callout__title'])
})`
  ${props => retrieveTheme('notifications.title', props)}
`;

/** @component */
export default Title;

