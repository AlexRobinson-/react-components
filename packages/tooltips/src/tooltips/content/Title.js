import styled from 'styled-components';
import { zdFontWeightRegular } from '@zendesk/garden-css-variables';
import { retrieveTheme } from 'garden-react-theming';

const Title = styled.div`
  font-weight: ${zdFontWeightRegular};
  margin-bottom: 10px;

  ${props => retrieveTheme('tooltip.title', props)}
`;

export default Title;
