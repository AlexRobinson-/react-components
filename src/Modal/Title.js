import React, { Component, PropTypes } from 'react'

import styles from './styles.css'

export default class Header extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired
  }

  render () {
    const { children } = this.props

    return (
      <h1 className={ styles.header }>
        { children }
      </h1>
    )
  }
}
