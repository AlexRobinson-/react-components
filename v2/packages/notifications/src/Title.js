import styled from 'styled-components';
import classNames from 'classnames';
import CalloutStyles from '@zendesk/garden-css-callouts';
import { utils } from '@zendeskgarden/react-theming';

const Title = styled.div.attrs({
  className: classNames(CalloutStyles['c-callout__title'])
})`
  ${props => utils.retrieveTheme('callout.title', props)}
`;

/** @component */
export default Title;

