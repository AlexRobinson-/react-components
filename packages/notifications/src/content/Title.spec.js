import React from 'react';
import { shallow } from 'enzyme';
import Title from './Title';

it('should render default title styling', () => {
  const wrapper = shallow(<Title />)
  expect(wrapper).toMatchSnapshot();
});
