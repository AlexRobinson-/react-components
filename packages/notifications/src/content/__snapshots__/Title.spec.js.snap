// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render default title styling 1`] = `
<div
  className="Title-kEkEAm c-callout__title hsTLhl"
/>
`;
